### PR TITLE
[pylint] Type format incorrect check

### PR DESCRIPTION
**Pull request by l0lawrence**
fixes https://github.com/Azure/azure-sdk-for-python/issues/22308

### CODE REVIEW

File: tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 1]
```diff
@@ -1397,6 +1408,23 @@ def check_parameters(self, node):
                 msgid="docstring-keyword-should-match-keyword-only", args=(", ".join(missing_kwonly_args)), node=node, confidence=None
             )
 
+        type_format = [":class", "Union", "Optional", "Dict", "List", "Callable", "None", "Iterable", "Tuple", "Mapping", "Any"]
```
Comment (l0lawrence) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:1411]: incorrect wording to find within a docstring

File: tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 2]
```diff
@@ -1397,6 +1411,22 @@ def check_parameters(self, node):
                 msgid="docstring-keyword-should-match-keyword-only", args=(", ".join(missing_kwonly_args)), node=node, confidence=None
             )
 
+        # check that all types are formatted correctly
+        for keyword, doc_type in docstring_keyword_args.items():
```
Comment (MarcelFox) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:1415]: I think it'll would be more efficient to use _list comprehension_ logic for both _for cases_.

File: tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 3]
```diff
@@ -1449,6 +1469,13 @@ def check_return(self, node):
                 has_return = True
             if line.startswith("rtype"):
                 has_rtype = True
+                try:
+                    if line.split("rtype")[1] and ":class" in line.split("rtype")[1]:
```
Comment (kristapratico) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:1473]: could we just do?

```suggestion
                    if ":class" in line.split("rtype")[1]:
```

File: tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py [hunk 4]
```diff
@@ -1449,6 +1469,13 @@ def check_return(self, node):
...
+                            self.add_message(
+                                msgid="docstring-type-do-not-use-class", args=(line.split("rtype")[1]), node=node, confidence=None
+                            )
+                except:
+                    pass
```
Comment (kristapratico) [inline tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:1478]: Do we need to check for :class: in :raises:?

**Review by kristapratico — COMMENTED** [review]
Can we add a few tests?

File: tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py (hunk 4 above)
Comment (l0lawrence) [reply tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py:1474]: Just double checked: ` you can use the :class: role to link to a class object in the Raises section, as long as the class is defined in the same module or imported from another module.`

**Review by kristapratico — APPROVED** [review]

**Pull request merged**